### PR TITLE
Ret 169 za bring up

### PR DESCRIPTION
**Pull request by sotaoverride**
ZA stage (head) bring up.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ build-cross/
 stm32-tools/
 build-host/
 cmake-build-debug/
+node_modules
```
Comment (sfoster1) [inline .gitignore:12]: why do we need this?

File: head/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# head source tree
+
+# add_subdirectory(src)
```
Comment (sfoster1) [inline head/CMakeLists.txt:3]: delete?

File: head/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+if (${CMAKE_CROSSCOMPILING})
```
Comment (sfoster1) [inline head/CMakeLists.txt:5]: don't seem to be adding `core` anywhere

**Review by sfoster1 — CHANGES_REQUESTED** [review]
We should
- Get rid of the uart stuff
- Clean up some commented out stuff in the cmake
- Add CI in github actions

File: .gitignore (hunk 1 above)
Comment (sfoster1) [reply .gitignore:12]: oh that makes sense i guess

File: head/core/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# This CMakeLists.txt handles compiling all the parts of the head
+# module that are portable between host and cross compilation as a static
+# library. It is included in both host and cross configurations.
+
+add_library(head-core STATIC
+  ${CMAKE_CURRENT_SOURCE_DIR}/
```
Comment (sfoster1) [inline head/core/CMakeLists.txt:6]: you can't really do this, you need to specify files. and as you've probably noticed from the failing builds, it needs a file to also determine the language

**Review by amitlissack — APPROVED** [review]
Looks good! 

Tested with CAN analyzer and FW happily responded to device info request.

File: head/core/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+add_library(head-core STATIC
+        head.cpp
+        )
+
+set_target_properties(head-core
+  PROPERTIES CXX_STANDARD 20
+             CXX_STANDARD_REQUIRED TRUE)
+
+target_include_directories(head-core
+  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include)
```
Comment (Laura-Danielle) [inline head/core/CMakeLists.txt:14]: This is also the wrong path for the include directory -- if you copied from gantry that project hasn't been updated in awhile. The path specified here would point to an include directory inside of the `head` subproject.

```
target_include_directories(head
        PUBLIC ${CMAKE_SOURCE_DIR}/include)
```

File: head/tests/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# this CMakeLists.txt file is only used when host-compiling to build tests
+find_package(Catch2 REQUIRED)
+include(CTest)
+include(Catch)
+include(AddBuildAndTestTarget)
+
+add_executable(
+        head
+        test_main.cpp
+        test_linear_motion_config.cpp
```
Comment (Laura-Danielle) [inline head/tests/CMakeLists.txt:10]: I don't know if we need to run this test file multiple times.

File: include/head/core/communication.hpp [hunk 7]
```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <array>
+#include <span>
+
+#include "common/core/bit_utils.hpp"
+#include "common/core/io.hpp"
```
Comment (Laura-Danielle) [inline include/head/core/communication.hpp:7]: This is leftover from a uart task which we don't have anymore. We should delete this file.

File: include/head/core/head_messages.hpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <variant>
+
+#include "motor-control/core/motor_messages.hpp"
```
Comment (Laura-Danielle) [inline include/head/core/head_messages.hpp:5]: Same with this file. It should be deleted.

File: head/core/CMakeLists.txt (hunk 5 above)
Comment (sotaoverride) [reply head/core/CMakeLists.txt:14]: So there is an include directory at root level, with sub directories for can, pipettes etc in it. Should we get rid of that and each subproject would have an include of their own nested inside of it? I figured the idea was for all subprojects to use a common include..

File: head/core/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# This CMakeLists.txt handles compiling all the parts of the head
+# module that are portable between host and cross compilation as a static
+# library. It is included in both host and cross configurations.
+
+add_library(head-core STATIC
+        head.cpp
```
[the remark this answers is not in the document]
Comment (sotaoverride) [reply head/core/CMakeLists.txt:6]: yes, it's on the same level as the CMakeLists.txt , so I think thats how this just works.

File: head/core/CMakeLists.txt (hunk 5 above)
Comment (sotaoverride) [reply head/core/CMakeLists.txt]: thanks, finally makes sense to me. [0230e63](https://github.com/Opentrons/ot3-firmware/pull/104/commits/0230e63cf179970f6c151b24b1296bbe0661a901)

**Pull request merged**